src/constitutive/TACSBladeStiffenedShellConstitutive.cpp

```diff
@@ -2646,15 +2705,166 @@ TacsScalar TACSBladeStiffenedShellConstitutive::computeCriticalShearLoad(
+                             &dfdN1Local, &dfdN1CritLocal, &dfdN12Local,
+                             &dfdN12CritLocal);
+  // dfdN1Local *= dKSdf[2];
+  // dfdN12Local *= dKSdf[2];
+  // dfdN1CritLocal *= dKSdf[2];
+  // dfdN12CritLocal *= dKSdf[2];
+
+  // Convert sensitivity w.r.t applied loads into sensitivity w.r.t DVs
+  TacsScalar dfdPanelStress[NUM_STRESSES];
+  memset(dfdPanelStress, 0, NUM_STRESSES * sizeof(TacsScalar));
+  dfdPanelStress[0] = -dfdN1Local;
+  dfdPanelStress[2] = dfdN12Local;
+  this->addPanelStressDVSens(scale, strain, dfdPanelStress,
+                             &dfdx[this->panelDVStartNum]);
+
+  // Convert the sensitivities of  the critical loads w.r.t the D matrix
+  // entries to sensitivities of the buckling failure criteria w.r.t the D
+  // matrix entries
+  TacsScalar localBucklingDMatSens[4];
+  for (int ii = 0; ii < 4; ii++) {
+    localBucklingDMatSens[ii] =
+        N1CritSens[ii] * dfdN1CritLocal + N12CritSens[ii] * dfdN12CritLocal;
+  }
+
+  // --- Panel thickness sensitivity ---
+  if (this->panelThickNum >= 0) {
+    const int dvNum = this->panelThickLocalNum;
+    // --- Local buckling contribution ---
+    const TacsScalar dMatdt = 0.25 * t * t;  // d/dt(t^3/12) = t^2/4
+    for (int ii = 0; ii < this->numPanelPlies; ii++) {
+      TacsScalar* Q = &(this->panelQMats[ii * NUM_Q_ENTRIES]);
+      dfdx[dvNum] +=
+          scale * dMatdt * this->panelPlyFracs[ii] *
+          (localBucklingDMatSens[0] * Q[0] + localBucklingDMatSens[1] * Q[3] +
+           localBucklingDMatSens[2] * Q[1] + localBucklingDMatSens[3] * Q[5]);
+    }
+  }
+
+  // --- Panel Ply fraction sensitivities ---
+  for (int ii = 0; ii < this->numPanelPlies; ii++) {
+    if (this->panelPlyFracNums[ii] >= 0) {
+      const int dvNum = this->panelPlyFracLocalNums[ii];
+      // --- Local buckling contribution ---
+      const TacsScalar DFactor = t * t * t / 12.0;
+      const TacsScalar* const Q = &(this->panelQMats[ii * NUM_Q_ENTRIES]);
+      // Do df/dx += df/dD11 * dD11/dx + df/dD22 * dD22/dx + df/dD12 * dD12/dx
+      // + df/dD66 * dD66/dx
+      dfdx[dvNum] +=
+          scale * DFactor *
+          (localBucklingDMatSens[0] * Q[0] + localBucklingDMatSens[1] * Q[3] +
+           localBucklingDMatSens[2] * Q[1] + localBucklingDMatSens[3] * Q[5]);
+    }
+  }
+
+  // --- Stiffener pitch sensitivity ---
+  if (this->stiffenerPitchNum >= 0) {
+    dfdx[this->stiffenerPitchLocalNum] +=
+        scale *
+        (N12CritSens[4] * dfdN12CritLocal + N1CritSens[4] * dfdN1CritLocal);
+  }
+}
+
+TacsScalar
+TACSBladeStiffenedShellConstitutive::computeCriticalLocalAxialLoadSens(
+    const TacsScalar D11, const TacsScalar D22, const TacsScalar D12,
+    const TacsScalar D66, const TacsScalar L, TacsScalar* const D11Sens,
+    TacsScalar* const D22Sens, TacsScalar* const D12Sens,
+    TacsScalar* const D66Sens, TacsScalar* const LSens) {
+  const double pi2 = M_PI * M_PI;
```

Comment:
If you look at where this method gets called in `addLocalPanelBucklingDVSens`, the value of `dfN1crit` isn't known yet because it relies on the value of `N1Crit`, which is computed by this method.

I agree that the ordering of operations in the buckling sensitivities could be improved quite a bit, but it doesn't seem to have a significant effect on the overall computation time so I will just leave it as is for now.